openstack-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1009 @@
+      + table.standardOptions.withOverridesMixin([
+        table.fieldOverride.byName.new('Enabled')
+        + table.fieldOverride.byName.withProperty('custom.displayMode', 'color-text')
+        + table.fieldOverride.byName.withPropertiesFromOptions(
+          table.standardOptions.withMappings(
+            {
+              type: 'value',
+              options: {
+                'false': {
+                  color: 'red',
+                  index: 1,
+                  text: 'False',
+                },
+                'true': {
+                  color: 'green',
+                  index: 0,
+                  text: 'True',
+                },
+              },
+            }
+          ),
+        ),
+      ])
+      + table.queryOptions.withTransformationsMixin([
+        {
+          id: 'joinByField',
+          options: {
+            byField: 'Time',
+            mode: 'outer',
+          },
+        },
+        {
+          id: 'organize',
+          options: {
+            excludeByName: {
+              Time: true,
+              Value: true,
+              __name__: true,
+              description: true,
+              job: true,
+              parent_id: true,
+            },
+            includeByName: {},
+            indexByName: {
+              Time: 3,
+              Value: 11,
+              __name__: 4,
+              description: 5,
+              domain_id: 7,
+              enabled: 6,
+              id: 1,
+              instance: 2,
+              is_domain: 8,
+              job: 10,
+              name: 0,
+              parent_id: 9,
+            },
+            renameByName: {
+              domain_id: 'Domain ID',
+              enabled: 'Enabled',
+              id: 'ID',
+              instance: 'Instance',
+              is_domain: 'Is domain',
+              name: 'Name',
+            },
+          },
+        },
+      ]),
+
+    vms:
+      commonlib.panels.generic.timeSeries.base.new(
+        'VMs',
+        targets=[t.vms],
+        description='The current number of total and running virtual machines.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
+    instanceUsage:
+      commonlib.panels.generic.timeSeries.percentage.new(
+        'Instance usage',
+        targets=[t.instanceUsage],
+        description='Percentage of the maximum number of instances in use for each project.'
+      )
+      + timeSeries.standardOptions.withUnit('percentunit')
+      + timeSeries.standardOptions.withMax(1)
+      + timeSeries.standardOptions.withMin(0),
+
+    vCPUUsage:
+      commonlib.panels.generic.timeSeries.percentage.new(
+        'VCPU usage',
+        targets=[t.vCPUUsage],
+        description='Percentage of the maximum number of virtual CPUs in use for each project.'
+      )
+      + timeSeries.standardOptions.withUnit('percentunit')
+      + timeSeries.standardOptions.withMax(1)
+      + timeSeries.standardOptions.withMin(0),
+
+    memoryUsage:
```

Comment:
This may be better suited with a right-aligned table? Currently a single instance and five projects are already crowding the legend.

![image](https://github.com/grafana/jsonnet-libs/assets/1282515/3eb1f014-1ba2-4e65-94bd-5a764d6264c9)